### PR TITLE
refactor retinanet

### PR DESCRIPTION
**Pull request by xchani**
1. move shared parameter into initial function for convinence
2. replace list by dict for return value type of retina neck

### CODE REVIEW

File: models/retinanet/builder.py [hunk 1]
```diff
@@ -323,19 +326,18 @@ def get_prediction(self, conv_feat, im_info):
...
+            cls_score = X.sigmoid(data=cls_logit_dict["stride%s" % s])
+            bbox_delta = bbox_delta_dict.pop("stride%s" % s)
 
-            cls_logit_dict.update({'cls_logit_stride%s' % s: cls_logit})
-            bbox_delta_dict.update({'bbox_delta_stride%s' % s: bbox_delta})
+            cls_score_dict["cls_score_stride%s" % s] = cls_score
+            bbox_delta_dict["bbox_delta_stride%s" % s] = bbox_delta
 
-        args_dict = {**cls_logit_dict, **bbox_delta_dict}
+        args_dict = {**cls_score_dict, **bbox_delta_dict}
```
Comment (RogerChern) [inline models/retinanet/builder.py]: Redundant temp variable definition.

File: models/retinanet/builder.py [hunk 2]
```diff
@@ -494,7 +500,15 @@ def get_retinanet_neck(data):
             name="P7_conv"
         )
 
-        return p3_conv, p4_conv, p5_conv, P6, P7
+        self.neck = dict(
+            stride8=p3_conv,
+            stride16=p4_conv,
+            stride32=p5_conv,
+            stride64=P6,
```
Comment (RogerChern) [inline models/retinanet/builder.py]: P6 -> p6, P7 -> p7.

File: models/retinanet/metric.py [hunk 3]
```diff
@@ -1,25 +1,42 @@
...
-        # pred_label = mx.nd.argmax_channel(pred)
-        pred_label = pred_label.asnumpy().astype('int32')
         # label (b, p)
         label = label.asnumpy().astype('int32')
-
         keep_inds = np.where(label >= 1)
+
+        if pred.shape[-1] == 1:
+            # treat as foreground if score larger than 0.05
+            pred_label = pred[:, :, 0] > 0.05
```
Comment (RogerChern) [inline models/retinanet/metric.py]: 0.5 is the proper threshold.

File: models/retinanet/metric.py [hunk 4]
```diff
@@ -1,25 +1,42 @@
...
+            label = label[keep_inds]
+        else:
+            # pred (b, p, c) , c is 0 ~ CLASSNUM-1
+            pred_label = mx.nd.argmax(pred, axis=2)
```
Comment (RogerChern) [inline models/retinanet/metric.py]: max score AND score > 0.5 is propoer.

File: models/retinanet/metric.py (hunk 3 above)
Comment (RogerChern) [reply models/retinanet/metric.py]: Or maybe add a stricter variant of this metric.

File: models/retinanet/builder.py (hunk 1 above)
Comment (xchani) [reply models/retinanet/builder.py]: used in decode_retina below

Comment (RogerChern) [reply models/retinanet/builder.py]: `Decode_retina` is the only place where the temp var `args_dict` is used. It is better to directly give `cls_score_dict` and `bbox_delta_dict` to `decode_retina`.

**Pull request merged**